app/models/book.rb

```diff
@@ -5,11 +5,22 @@ class Book < ApplicationRecord
   validates :title, presence: true, uniqueness: { case_sensitive: false}
   validates :author, presence: true
 
-  def self.available
-    includes(:checkouts).where(checkouts: { id: nil })
+  def self.unavailable
+    Book.joins(:checkouts).where(checkouts: {closed_at: nil })
   end
 
-  def self.unavailable
-    includes(:checkouts).where.not(checkouts: { id: nil })
+  def self.available
+    # I needed to use a LEFT JOIN because
+    # `Book.joins(:checkouts).where.not(checkouts: {closed_at: nil })`
+    # displayed checkouts where ANY checkout associated with the book
+    # and user would be listed as unavailable. I need books to listed as
+    # available when EVERY checkout is closed. 
+    Book.joins(
```

Comment:
Thanks for the comment. I would probably add that to the commit message rather than in the code itself. More details here: https://robots.thoughtbot.com/letting-your-code-speak-for-itself

Also you may be able to use these new methods in Rails 5 instead of dropping down to raw SQL: http://blog.bigbinary.com/2016/03/24/support-for-left-outer-joins-in-rails-5.html